tensorflow-examples/src/main/java/org/tensorflow/model/examples/mnist/training/LinearRegressionExample.java

```diff
@@ -0,0 +1,154 @@
+        }
+
+        try (Graph graph = new Graph()) {
+            Ops tf = Ops.create(graph);
+
+            // Define placeholders
+            Placeholder<TFloat32> X = tf.placeholder(TFloat32.DTYPE, Placeholder.shape(Shape.scalar()));
+            Placeholder<TFloat32> Y = tf.placeholder(TFloat32.DTYPE, Placeholder.shape(Shape.scalar()));
+
+            // Define variables
+            Variable<TFloat32> weight = tf.variable(Shape.scalar(), TFloat32.DTYPE);
+            Variable<TFloat32> bias = tf.variable(Shape.scalar(), TFloat32.DTYPE);
+
+            // Init variables
+            Assign<TFloat32> weightInit = tf.assign(weight, tf.constant(1f));
+            Assign<TFloat32> biasInit = tf.assign(bias, tf.constant(1f));
+
+            // Define the model function weight*x + bias
+            Mul<TFloat32> mul = tf.math.mul(X, weight);
+            Add<TFloat32> yPredicted = tf.math.add(mul, bias);
+
+            // Define loss function MSE
+            Pow<TFloat32> sum = tf.math.pow(tf.math.sub(yPredicted, Y), tf.constant(2f));
+            Div<TFloat32> mse = tf.math.div(sum, tf.constant(2f * N));
+
+            // Define gradient operators
+            List<Variable<TFloat32>> variables = new ArrayList<>();
+            variables.add(weight);
+            variables.add(bias);
+
+            Gradients gradients = tf.gradients(mse, variables);
+
+            Constant<TFloat32> alpha = tf.constant(0.2f);
+
+            ApplyGradientDescent<TFloat32> weightGradientDescent = tf.train.applyGradientDescent(weight, alpha, gradients.dy(0));
+            ApplyGradientDescent<TFloat32> biasGradientDescent = tf.train.applyGradientDescent(bias, alpha, gradients.dy(1));
+
+            try (Session session = new Session(graph)) {
+                // Initialize graph variables
+                session.runner()
+                        .addTarget(weightInit)
+                        .addTarget(biasInit)
+                        .run();
+
+                // Train the model on data
+                for (int i = 0; i < xValues.length; i++) {
+                    float y = yValues[i];
+                    float x = xValues[i];
+
+                    Tensor<TFloat32> xTensor = TFloat32.scalarOf(x);
+                    Tensor<TFloat32> yTensor = TFloat32.scalarOf(y);
+
+                    session.runner()
+                            .addTarget(weightGradientDescent)
+                            .addTarget(biasGradientDescent)
+                            .feed(X.asOutput(), xTensor)
+                            .feed(Y.asOutput(), yTensor)
+                            .run();
+
+                    System.out.println("Training phase");
+                    System.out.println("X is " + x + " Y is " + y);
+                }
+
+                // Extract the weight value
+                Tensor<TFloat32> weightValue = session.runner()
+                        .fetch("Variable")
+                        .run().get(0).expect(TFloat32.DTYPE);
```

Comment:
same thing, fetched tensors must be enclosed in a `try-with-resource` block. You probably want to fetch both values in one run as well, i.e. add `fetch("Variable_1")` in this run as well, then access it with index `[1]`.

By the way, for better readability, when you create your variable, you should give them a proper name to use for fetching, like `tf.withName("weight").variable(tf.constant(1))`. Then you fetch it with `.fetch("weight")`